src/parser.js

```diff
@@ -0,0 +1,143 @@
+const yargs = require('yargs/yargs');
+const {hideBin} = require('yargs/helpers');
+const moment = require('moment');
+
+const errorMsg = {
+  ID_LENGTH: 'ID must be a string with a length of 16',
+  DATE_FMT: 'Invalid date. Provide a date in YYYY-MM-DD HH:MM:SS format.',
+  OPTION_AMOUNT: 'At least one option is required',
+  EMPTY_REQ: 'Empty request. Please input a command',
+};
+
+const arg = yargs(hideBin(process.argv))
+    .command('ls', 'Output the list of tasks', (yargs) => {
+      yargs
+          .group(['due', 'overdue', 'completed'], 'Filter Options:')
+          .option('due', {
+            describe: 'Filter tasks to show only pending tasks',
+            alias: 'd',
+            type: 'boolean',
+            conflicts: ['overdue', 'completed'],
+          })
+          .option('overdue', {
+            describe: 'Filter tasks to show only overdue tasks',
+            alias: 'o',
+            type: 'boolean',
+            conflicts: ['due', 'completed'],
+          })
+          .option('completed', {
+            describe: 'Filter tasks to show only completed tasks',
+            alias: 'c',
+            type: 'boolean',
+            conflicts: ['due', 'overdue'],
+          })
+          .strictOptions();
+    }, function(arg) {
+      // TODO: run the function
+      console.log(arg);
```

Comment:
My preference is to throw an Error if such functionality is unimplemented. You can trace which command is run by checking the error message.
```suggestion
      throw Error('ls called, but is not implemented yet');
```